tests/cupy_tests/math_tests/test_arithmetic.py

```diff
@@ -66,66 +46,163 @@ def check_raises_with_numpy_input(self, nargs, name):
+        'name': ['floor_divide', 'fmod', 'remainder'],
+    }) + testing.product({
+        'arg1': ([numpy.array([-3, -2, -1, 1, 2, 3], dtype=d)
+                  for d in negative_no_complex_types]
+                 + [0, 0.0, 2, 2.0, -2, -2.0, True, False]),
+        'arg2': ([numpy.array([-3, -2, -1, 1, 2, 3], dtype=d)
+                  for d in negative_no_complex_types]
+                 + [0, 0.0, 2, 2.0, -2, -2.0, True, False]),
+        'name': ['floor_divide', 'fmod', 'remainder'],
+    })
+))
+class TestArithmeticBinary(unittest.TestCase):
 
-    def test_true_divide(self):
+    @testing.numpy_cupy_allclose(atol=1e-4)
+    def test_binary(self, xp):
+        arg1 = self.arg1
+        arg2 = self.arg2
+
+        # TODO(niboshi): Fix this: power(0, 1j)
+        #     numpy => 1+0j
+        #     cupy => 0j
+        if (self.name == 'power'
+                and numpy.asarray(arg1 == 0).any()
+                and numpy.asarray(arg2).dtype in complex_types):
+            return xp.array(True)
+
+        # TODO(niboshi): Fix this: xp.power(0j, 0)
+        #     numpy => 1+0j
+        #     cupy => 0j
+        if (self.name == 'power'
+                and numpy.asarray(arg1).dtype in complex_types
+                and numpy.asarray(arg1 == 0j).any()
+                and numpy.asarray(arg2 == 0).any()):
+            return xp.array(True)
+
+        # TODO(niboshi): Fix this: xp.add(0j, xp.array([2.], 'f')).dtype
+        #     numpy => complex64
+        #     cupy => complex128
+        if (isinstance(arg1, complex)
+                and (numpy.asarray(arg2).dtype
+                     in (numpy.float16, numpy.float32))):
+            return xp.array(True)
+
+        if isinstance(arg1, numpy.ndarray):
+            arg1 = xp.asarray(arg1)
+        if isinstance(arg2, numpy.ndarray):
+            arg2 = xp.asarray(arg2)
+
+        func = getattr(xp, self.name)
         with testing.NumpyError(divide='ignore'):
-            self.check_binary('true_divide')
-        self.check_raises_with_numpy_input(2, 'true_divide')
+            with numpy.warnings.catch_warnings():
+                numpy.warnings.filterwarnings('ignore')
+                y = func(arg1, arg2)
 
-    def test_true_divide_negative(self):
-        with testing.NumpyError(divide='ignore'):
-            self.check_binary_negative('true_divide')
+        # TODO(niboshi): Fix this
+        if xp is cupy:
+            if (xp.asarray(arg1).dtype in (numpy.float16, numpy.float32)
+                    and isinstance(arg2, complex)):
+                y = y.astype(numpy.complex64)
 
-    def test_floor_divide(self):
-        with testing.NumpyError(divide='ignore'):
-            self.check_binary('floor_divide', no_complex=True)
-        self.check_raises_with_numpy_input(2, 'floor_divide')
+        # TODO(niboshi): Fix this. Treat NaN and inf interchangeably
+        y = xp.asarray(y)
+        if y.dtype in (float_types + complex_types):
```

Comment:
What does it mean?

Reply:
Sorry, it was not clear. I added comments, and also narrowed the condition.

Reply:
Ok. Actually this behavior depends on CPU architecture I think (I mean ARM may returns different result for example).

Reply:
I see. So you mean we don't have to fix this and we can just ignore the difference in the test?

Reply:
IMO, no because in ARM architecture for example TX1, this test fails maybe.

Reply:
I'm sorry, it was not clear. I meant there are 2 choices and I think we should choose 2. Is that correct?

1. Fix `cupy.divide` to output `nan` or `inf` according to the CPU architecture
2. Do not fix `cupy.divide`. Instead fix the test to ignore the difference.

Reply:
Yes 2 is better i think. IMO, these behavior is undefined. So it is ok to skip its test.